thorlcr/activities/join/thjoinslave.cpp

```diff
@@ -299,7 +299,10 @@ class JoinSlaveActivity : public CSlaveActivity, implements ILookAheadStopNotify
             stopOtherInput();
             throw;
         }
-        asyncSecondaryStart.wait();
+        {
+            BlockedActivityTimer t(slaveTimerStats, timeActivities);
```

Comment:
not sure about this.

why is this blocked time?

input is on a thread - asynchronously starting.
Totalcycles at this point, will be the total time both took to start (overlapping).
The timing in queryLocalCycles() can go wrong because it is considering totalcycles - (input1-time+intput2-time), i.e. as if started synchronously.

This timer will be : input2 time - input1 time (or 0 if input2 was quicker).
Counting it as  lookahead time, means "processCycles" in queryLocalCycles() will be total+this lookahead time, and mean that when the sum total for both inputs are deduced it should be correct.

Whatever the conclusion, needs a good comment here explaining the logic clearly.